src/unitxt/operators.py

```diff
@@ -393,6 +393,11 @@ def prepare(self):
     def process(
         self, instance: Dict[str, Any], stream_name: Optional[str] = None
     ) -> Dict[str, Any]:
+        # Need to deep copy instance, because when assigning two dictionary fields,
+        # dict_set() the target field dictionary fields.
+        # This means that if this target field was assigned to another field before,
+        # the field is updated as well.
+        instance = deepcopy(instance)
```

Comment:
I think that perhaps this was a feature, not a bug, and perhaps better make this `deepcopy` depend on a special flag arg: `use_deepcopy`